src/Model.cc

```diff
@@ -135,6 +135,12 @@ void Model::start(AnimationKind kind)
 {
     m_kind = kind;
 
+    if (m_window) {
```

Comment:
visibility ref is conceptually a different thing. I suggest to store the visibility ref alongside with the model, i.e.

```
QMap<KWin::EffectWindow*, Model> m_models;
```

will become

```
struct AnimationData
{
    Model model;
    EffectWindowVisibileRef visibilityRef;
};
QMap<EffectWindow *, AnimationData> m_animations;
```

Reply:
Done ^^